### PR TITLE
Add support for Linux and macOS

### PR DESCRIPTION
**Pull request by Hexxonite**
Add fix for window-less operation, Add GlobalFont class to return a valid Default on multiple systems.

_EDIT: further discussion in #211_

### CODE REVIEW

**Comment by swharden** [comment]
hey @Hexxonite thanks for this PR! I'm going to use it to make modifications to the build pipeline to try to get it to build and run tests on Linux and macOS automatically.

Bear with me and my frequent commits while I try to appease the Azure Pipeline gods =]

File: src/ScottPlot/GlobalFont.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    public static class GlobalFont
+    {
+        /// <summary>
+        ///     This method will check for several preffered fonts
+        ///     to exist on the system and return the exact name.
+        /// </summary>
+        /// <returns></returns>
+        public static string GetDefault()
+        {
+            foreach (FontFamily font in FontFamily.Families)
```
Comment (StendProg) [inline src/ScottPlot/GlobalFont.cs]: In your implementation you always get `DEJAVU` font if `FontFamily.Families` in alphabetical order.
To make ordered font select, like at first `SEGOE`, at second `DEJAVU`, at third `SANS` you can use linq like below.
Looks not so clear, but always select `SEGOE` if available:
```c#
string[] PrefFonts = new string[]{"SEGOE", "DEJAVU", "SANS"};
var PrefSystemFont = PrefFonts
    .SelectMany( pf => FontsFamily.Families, (PrefFont, SystemFont) => new { PrefFont, SystemFont })
    .Where( pfsf => pfsf.SystemFont.Name.ToUpper().Contains(pfsf.PrefFont.Name))
    .Select(x => x.SystemFont.Name);   
if (PrefSystemFont.Any())
    return PrefSystem.First();
```

File: src/ScottPlot/GlobalFont.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            {
+                var fntu = font.Name.ToUpper();
+                if (fntu.Contains("SEGOE") || fntu.Contains("DEJAVU") || fntu.Contains("SANS"))
+                    return font.Name;
+            }
+            // Debug.WriteLine("No preffered Font found! Using any as fallback..");
+            foreach (FontFamily font in FontFamily.Families)
```
Comment (StendProg) [inline src/ScottPlot/GlobalFont.cs]: Looks more clear for me:
```c#
if ( !FontFamily.Families.Any() )
  throw new Exception("No Fonts found on the System!");

return FontFamily.Families.First().Name;
  
```

**Comment by swharden** [comment]
It's building on Linux and tests pass now 🎉🎉🎉

@StendProg thanks for your suggestions, your code is indeed clearer and I will make the changes soon!

Currently the build is failing on macOS. I'll see what I can do to make that succeed, then I'll refine the font code.

**Comment by Hexxonite** [comment]
Thanks for fixing this 👍 

I'm glad I could help in some way

File: src/ScottPlot/Config/TextLabel.cs [hunk 3]
```diff
@@ -10,27 +10,29 @@ namespace ScottPlot.Config
...
-        private Graphics gfx = Graphics.FromHwnd(IntPtr.Zero);
-
         public string text = "";
         public bool visible = true;
 
         public float fontSize = 12;
         public bool bold = false;
 
-        private string _fontName = "Segoe UI";
+        private string _fontName = GlobalFont.GetDefault();
```
Comment (StendProg) [inline src/ScottPlot/Config/TextLabel.cs]: This initialization not so simple now. It's better to move it to constructor.

**Comment by StendProg** [comment]
> It's building on Linux and tests pass now 🎉🎉🎉

Does this #211 covered with tesfor now? Failed without PR fix? No ideats  how to check, but very interesting.

**Comment by swharden** [comment]
@StendProg the configuration for building/testing is in `dev/devops/azure-pipelines.yml`

Since this PR branch has a modified `azure-pipelines.yml` which includes Linux, only this PR branch is being checked for Linux. After I merge it the new `azure-pipelines.yml` (which includes Linux) will be in the master branch, so all future commits (and future PRs) will get built and tested on Linux.

Currently I'm trying to figure out how to get macOS to build properly. Once I get all this building/testing stuff figured out, then I'll go back and work on integrating the font-related changes.

**Comment by swharden** [comment]
image

If you haven't looked at it before, you may find it interesting to look at the output of the build and tests.

* click "show all checks"
* click "details" by a build
* click "view more details on Azure Pipelines" at the bottom of the page

You can then see a list of tasks and the console output. When something fails, this console output shows which step failed. It's great to find line numbers of code causing exceptions.

**Comment by swharden** [comment]
After spending a whole lot of time messing with it I finally got Azure Pipelines to build ScottPlot and run tests on MacOS and Linux. Thanks @Hexxonite for initially encouraging me to do this!

I also combined @Hexxonite and @StendProg's suggestions together to make a new `ScottPlot.Config.Fonts` module. It contains these methods. @StendProg if you think LINQ would improve readability let me know how you would do it. Otherwise, I think it's looking pretty good!

https://github.com/swharden/ScottPlot/blob/d086ec0d94495d2199d46b771462401432805e00/src/ScottPlot/Config/Fonts.cs#L15-L24

https://github.com/swharden/ScottPlot/blob/d086ec0d94495d2199d46b771462401432805e00/src/ScottPlot/Config/Fonts.cs#L29-L39

**Pull request merged**